In [3]:
import pandas as pd
import folium

#reading our data into a dataframe
df = pd.read_csv("public_wifi_locations.csv")

#counts of zipcodes
zipcode_counts = df['zip'].value_counts()

#finding the center of the map whenever the locations are displayed
center = [df['y'].mean(),df['x'].mean()] #just mean of x and y coordinates

#creating the map
map = folium.Map(location=center, zoom_start=10)

# for each zipcode, mark it on the map
for zipcode, count in zipcode_counts.items():
    # x and y data for the coordinates of zipcodes
    zipcode_data = df[df['zip'] == zipcode].iloc[0]  # first occurance of zip
    x_coord = float(zipcode_data['x'])
    y_coord = float(zipcode_data['y'])
    popup_content = f"ZIP Code: {zipcode}<br>Count: {count}"
    folium.Marker([y_coord, x_coord], popup=popup_content).add_to(map)

#saving to html
map.save("zipcode_map.html")

print(zipcode_counts)

15104.0    18
15210.0    13
15219.0     9
15215.0     8
15203.0     6
15213.0     5
15235.0     5
15226.0     4
15216.0     4
15221.0     3
15217.0     3
15205.0     3
15227.0     3
15025.0     2
15218.0     2
15110.0     2
15137.0     2
15108.0     2
15236.0     2
15209.0     2
15212.0     2
15207.0     2
15202.0     2
15106.0     2
15220.0     2
15144.0     1
15143.0     1
15206.0     1
15116.0     1
15064.0     1
15129.0     1
15017.0     1
15136.0     1
15239.0     1
15071.0     1
15234.0     1
15044.0     1
15131.0     1
15241.0     1
15139.0     1
15238.0     1
15208.0     1
15237.0     1
15201.0     1
15228.0     1
15146.0     1
15204.0     1
15132.0     1
15120.0     1
15101.0     1
15065.0     1
15102.0     1
15233.0     1
Name: zip, dtype: int64


The metric used here was the number of public wifi locations. The way it was assigned was taking a zipcode and counting all of the public wifi locations that are in that zipcode. We used a python extension called folium to create a map with all of the locations on it and also a count on the number of public wifi locations in that zipcode. 

The top 3 zipcodes with the most Public Wifi Locations were
15104 - Braddock, PA  18
15210 - Mt Oliver   13
15219 - Downtown   9